### PR TITLE
Document the assoc arrays as per their string key.

### PR DESCRIPTION
**Pull request by dereuromark**
Follow https://github.com/cakephp/cakephp/pull/15811

With this almost all should be done now.

### CODE REVIEW

File: src/Console/HelpFormatter.php [hunk 1]
```diff
@@ -192,7 +192,7 @@ protected function _generateUsage(): string
     /**
      * Iterate over a collection and find the longest named thing.
      *
-     * @param array $collection The collection to find a max length of.
+     * @param array<\Cake\Console\ConsoleInputOption|\Cake\Console\ConsoleInputArgument|\Cake\Console\ConsoleInputSubcommand> $collection The collection to find a max length of.
```
Comment (othercorey) [inline src/Console/HelpFormatter.php:195]: Can you check this in 5.0 when merging?

**Comment by othercorey** [comment]
This one look ok too.

**Comment by dereuromark** [comment]
How can this be solved?
```
Error: Class Cake\Console\Shell is marked as deprecated (see https://psalm.dev/098)
Error: Class Cake\Console\Shell is marked as deprecated (see https://psalm.dev/098)
Error: Process completed with exit code 2.
```
The --set-baseline didn't work

**Comment by othercorey** [comment]
You have to add an entry to the deprecation section (don't forget to increase occurrences). 
```
<DeprecatedClass occurrences="4">
</DeprecatedClass>
```

The `<code>` should be the full type in the annotation.

**Comment by ADmad** [comment]
> The --set-baseline didn't work

You need to specify the filename, so `--set-baseline=psalm-baseline.xml`.

**Comment by dereuromark** [comment]
Yeah, I figured it out. Psalm could be smart enough to get them from the config, but meeh^^

**Pull request merged**